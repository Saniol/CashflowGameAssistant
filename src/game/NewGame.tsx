import React, {useState, useCallback} from 'react';
import {View, StyleSheet, Picker, Text, Button} from 'react-native';
import {useHistory} from 'react-router-native';
import Lang from '../lang/pl';
import Proffessions from '../data/professions';
import Colors from '../global/Colors';
import Backgrounds from '../global/Backgrounds';
import LayoutStyles from '../global/LayoutStyles';

const NewGameStyles = StyleSheet.create({
    picker: {
        marginBottom: 18,
    },
});

const NewGame: React.FC = () => {
    const defaultValue = Proffessions[0].type;
    const [proffession, setProffession] = useState(defaultValue);
    const history = useHistory();
    const goToDashboard = useCallback(() => {
        history.push('/home');
    }, [history]);

    return (
        <View style={LayoutStyles.container}>
            <Text style={LayoutStyles.header}>{Lang.newGameHeader}</Text>
            <Text>{Lang.newGameSelectProfession}</Text>
            <Picker
                testID="newGameProffessionPicker"
                selectedValue={proffession}
                style={[Backgrounds.background, NewGameStyles.picker]}
                onValueChange={itemValue => setProffession(itemValue)}>
                {Proffessions.map(proffessionDefinition => (
                    <Picker.Item
                        key={proffessionDefinition.type}
                        label={proffessionDefinition.name}
                        value={proffessionDefinition.type}
                    />
                ))}
            </Picker>
            <Button
                testID="newGameStartButton"
                title={Lang.newGameStart}
                color={Colors.dark}
                onPress={() => {
                    goToDashboard();
                }}
            />
        </View>
    );
};

export default NewGame;
