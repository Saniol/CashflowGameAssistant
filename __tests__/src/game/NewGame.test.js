import 'react-native';
import React from 'react';
import {createMemoryHistory} from 'history';
import {Router} from 'react-router-native';
import {mount} from 'enzyme';
import NewGame from '../../../src/game/NewGame';
import renderWithContexts from '../../common/renderWithContexts';
import getByTestId from '../../common/getByTestId';

describe('<NewGame>', () => {
    renderWithContexts(NewGame);

    describe('start button', () => {
        let history;
        let wrapper;
        let pushSpy;

        beforeEach(() => {
            history = createMemoryHistory();
            wrapper = mount(
                <Router history={history}>
                    <NewGame />
                </Router>,
            );
            pushSpy = jest.spyOn(history, 'push');
        });

        describe('clicked after selecting proffession', () => {
            const selectProffession = value => {
                const picker = getByTestId(
                    wrapper,
                    'newGameProffessionPicker',
                ).first();

                picker.props().onValueChange(value);

                wrapper.update();
            };
            const pressStartButton = () => {
                const startButton = getByTestId(
                    wrapper,
                    'newGameStartButton',
                ).first();

                startButton.props().onPress();

                wrapper.update();
            };
            it('should set proper data in application state', () => {});
            it('should navigate to /home', () => {
                pressStartButton();

                expect(pushSpy.mock.calls[0][0]).toBe('/home');
            });
        });
    });
});
