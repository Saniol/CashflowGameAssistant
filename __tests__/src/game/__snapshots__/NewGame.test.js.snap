// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewGame> renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "marginTop": 12,
      "paddingHorizontal": 24,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 24,
        "fontWeight": "bold",
        "marginBottom": 12,
      }
    }
  >
    Rozpocznij nową grę
  </Text>
  <Text>
    Wybierz zawód:
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FCE9C4",
        },
        Object {
          "marginBottom": 18,
        },
      ]
    }
  >
    <RCTPicker
      items={
        Array [
          Object {
            "label": "Pilot linii lotniczych",
            "textColor": undefined,
            "value": "AirlinePilot",
          },
          Object {
            "label": "Lekarz",
            "textColor": undefined,
            "value": "Doctor",
          },
          Object {
            "label": "Inżynier",
            "textColor": undefined,
            "value": "Engineer",
          },
          Object {
            "label": "Dozorca",
            "textColor": undefined,
            "value": "Janitor",
          },
          Object {
            "label": "Prawnik",
            "textColor": undefined,
            "value": "Lawyer",
          },
          Object {
            "label": "Menadżer",
            "textColor": undefined,
            "value": "Manager",
          },
          Object {
            "label": "Mechanik",
            "textColor": undefined,
            "value": "Mechanic",
          },
          Object {
            "label": "Pielęgniarka",
            "textColor": undefined,
            "value": "Nurse",
          },
          Object {
            "label": "Policjant",
            "textColor": undefined,
            "value": "Policeman",
          },
          Object {
            "label": "Sekretarka",
            "textColor": undefined,
            "value": "Secretary",
          },
          Object {
            "label": "Nauczyciel",
            "textColor": undefined,
            "value": "Teacher",
          },
          Object {
            "label": "Kierowca ciężarówki",
            "textColor": undefined,
            "value": "TruckDriver",
          },
        ]
      }
      onChange={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
      testID="newGameProffessionPicker"
    />
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="newGameStartButton"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#EEB35F",
            },
          ]
        }
      >
        Rozpocznij
      </Text>
    </View>
  </View>
</View>
`;
